modules/wallet/database.go

```diff
@@ -1,21 +1,51 @@
 package wallet
 
 import (
+	"encoding/binary"
+	"reflect"
+
 	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/modules"
 	"github.com/NebulousLabs/Sia/types"
+
 	"github.com/NebulousLabs/bolt"
 )
 
 var (
+	// bucketHistoricClaimStarts maps a SiafundOutputID to the value of the
+	// siafund pool when the output was processed. It stores every such output
+	// in the blockchain. The wallet uses this mapping to determine the "claim
+	// start" value of siafund outputs in ProcessedTransactions.
 	bucketHistoricClaimStarts = []byte("bucketHistoricClaimStarts")
-	bucketHistoricOutputs     = []byte("bucketHistoricOutputs")
-	bucketSiacoinOutputs      = []byte("bucketSiacoinOutputs")
-	bucketSiafundOutputs      = []byte("bucketSiafundOutputs")
-	bucketSpentOutputs        = []byte("bucketSpentOutputs")
+	// bucketHistoricOutputs maps a generic OutputID to the number of siacoins
+	// the output contains. The output may be a siacoin or siafund output.
+	// Note that the siafund value here is not the same as the value in
+	// bucketHistoricClaimStarts; see the definition of SiafundOutput in
+	// types/transactions.go for an explanation. The wallet uses this mapping
+	// to determine the value of outputs in ProcessedTransactions.
+	bucketHistoricOutputs = []byte("bucketHistoricOutputs")
+	// bucketProcessedTransactions stores ProcessedTransactions in
+	// chronological order. Only transactions relevant to the wallet are
+	// stored. The key of this bucket is an autoincrementing integer.
+	bucketProcessedTransactions = []byte("bucketProcessedTransactions")
+	// bucketSiacoinOutputs maps a SiacoinOutputID to its SiacoinOutput. Only
+	// outputs that the wallet controls are stored. The wallet uses these
+	// outputs to fund transactions.
+	bucketSiacoinOutputs = []byte("bucketSiacoinOutputs")
+	// bucketSiacoinOutputs maps a SiafundOutputID to its SiafundOutput. Only
+	// outputs that the wallet controls are stored. The wallet uses these
+	// outputs to fund transactions.
+	bucketSiafundOutputs = []byte("bucketSiafundOutputs")
+	// bucketSpentOutputs maps an OutputID to the height at which it was
+	// spent. Only outputs spent by the wallet are stored. The wallet tracks
+	// these outputs so that it can reuse them if they are not confirmed on
+	// the blockchain.
+	bucketSpentOutputs = []byte("bucketSpentOutputs")
```

Comment:
ah, ok, I was confused because I didn't see the outputs being deleted anywhere. That seems slightly better to me, since otherwise `spentOutputs` will grow forever.